source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -17,6 +19,26 @@ const std::string& PerConnectionRateLimiter::key() {
   CONSTRUCT_ON_FIRST_USE(std::string, "per_connection_local_rate_limiter");
 }
 
+LocalRateLimitRequestDescriptorsQueue::LocalRateLimitRequestDescriptorsQueue()
+    : request_descriptors_queue_(
+          std::queue<absl::optional<std::vector<RateLimit::LocalDescriptor>>>()) {}
+
+void LocalRateLimitRequestDescriptorsQueue::push(
+    absl::optional<std::vector<RateLimit::LocalDescriptor>> request_descriptors) {
```

Comment:
I Imagine we don't need to have optional here. Do you consider just avoiding `push(abls::nullopt)`?

Reply:
We still need an optional here. In any request the filter is not enabled, we will not attach x-ratelimit-* headers.